api/src/types/items.ts

```diff
@@ -11,14 +6,10 @@ export type Item = Record<string, any>;
 
 export type PrimaryKey = string | number;
 
-export type Alterations = {
-	create: {
-		[key: string]: any;
-	}[];
-	update: {
-		[key: string]: any;
-	}[];
-	delete: (number | string)[];
+export type Alterations<T extends Item = Item, K extends keyof T | undefined = undefined> = {
+	create: Partial<T>[];
+	update: (K extends keyof T ? Partial<T> & Pick<T, K> : Partial<T>)[];
+	delete: (K extends keyof T ? T[K] : PrimaryKey)[];
```

Comment:
This type is missing the possibility that the update/create array's also take direct primary keys (so combined with the above simplification that would be):
```suggestion
export type Alterations<T extends Item = Item> = {
	create: (Partial<T> | PrimaryKey)[];
	update: (Partial<T> | PrimaryKey)[];
	delete: PrimaryKey[];
```